src/util/signature_utils.js

```diff
@@ -230,7 +231,13 @@ function _aws_request(req, region, service) {
             headers_for_sdk[sdk_key] = value;
         }
     }
+    // if req.body exists, convert it to x-www-form-urlencoded
+    let body;
+    if (req.body && service === 'sts') {
```

Comment:
so do we need to add a body for all sts actions or just to assume role? can you maybe share a link for why this is the needed solution?
Also, doesn't this affect non cli requests? how things are not failing on the sts sdk tests?